libyul/backends/wasm/WasmCodeTransform.cpp

```diff
@@ -165,30 +170,32 @@ wasm::Expression WasmCodeTransform::operator()(Identifier const& _identifier)
 
 wasm::Expression WasmCodeTransform::operator()(Literal const& _literal)
 {
-	u256 value = valueOfLiteral(_literal);
-	yulAssert(value <= numeric_limits<uint64_t>::max(), "Literal too large: " + value.str());
-	return wasm::Literal{static_cast<uint64_t>(value)};
+	return makeLiteral(translatedType(_literal.type), valueOfLiteral(_literal));
 }
 
 wasm::Expression WasmCodeTransform::operator()(If const& _if)
 {
-	// TODO converting i64 to i32 might not always be needed.
+	yul::Type conditionType = m_typeInfo.typeOf(*_if.condition);
+	yulAssert(m_dialect.types.count(conditionType) == 1, "");
 
 	vector<wasm::Expression> args;
 	args.emplace_back(visitReturnByValue(*_if.condition));
-	args.emplace_back(wasm::Literal{static_cast<uint64_t>(0)});
+	args.emplace_back(makeLiteral(translatedType(conditionType), 0));
```

Comment:
Is it possible to get rid of the `!= 0` check for some of the types?

Reply:
Yeah, I think so. Boolean values in wasm are always `i32` so as long as `_if.condition` is of the same type, I can skip the comparison. I just need to make sure that `true` is not required to be exactly 1 in wasm (I think anything non-zero is considered `true`).

Reply:
Sounds good!

Reply:
Done.